3rd-party-scripts/go-server/README.md

```diff
@@ -0,0 +1,16 @@
+# Go TLS self-server
+
+This minimal Go program lacks configuration options and relies on the standard
+library's http.ListenAndServeTLS handler. No external dependencies are
+required, making this portable across Go runtime variants (BoringCrypto, RHEL
+with OpenSSL, &c).
+
+To test the Go TLS server, you first need to install a Go toolchain
+environment (e.g., `sudo dnf install -y go`). Then, in this directory,
+run `go build`.
+
+Finally, to start a self-server with default configuration:
```

Comment:
what about certificates?

Reply:
Good idea.

Did you want me to support more of the [`tls.Config`](https://pkg.go.dev/crypto/tls#Config) options? It'd increase the size of the code a little, but bring more parity/flexibility between it and NSS/OpenSSL/... if y'all wanted to start testing against it.

Reply:
Having ability to specify them in the first place, so that one can run against RSA, EdDSA and ECDSA server would be the first step, if there's need for more options we can add them later.

Side note: the very detailed and extensive documentation on how to set up the servers is in the wiki: https://github.com/tlsfuzzer/tlsfuzzer/wiki/Server-setup

Reply:
Ah sorry, that's already possible from here so its just a documentation issue:

```
	cert := flag.String("cert", "localhost.crt", "Server certificate in PEM format")
	key := flag.String("key", "localhost.key", "Server certificate's corresponding key, in PEM format")
```

which should support RSA/ECDSA/EdDSA keys. Go doesen't support the legacy DSA keys any more (in upstream-supported versions).

The part I was wondering about is if I should also add flags like `-ciphers` and `-protocol_version` (and friends) to configure TLSv1.2/TLSv1.3/... options as well.

Reply:
Ok, then please document those cert/key options.

Ciphers would be rather useful, as I assume that go actually supports more ciphers than it enables by default. Protocol versions are less important, as we mostly care about TLS 1.2 and TLS 1.3 only anyway...

Reply:
FYI, I wrote [this](https://github.com/ueno/go-tlsdriver) for downstream testing some time ago and the server command accepts multiple certs/keys, as well as curves, protocol versions, etc.

Reply:
Sorry about the delay, been busy at work :-) 

I'm fine with @ueno's, it looks a lot more feature complete than mine.

Reply:
Done https://github.com/tlsfuzzer/tlsfuzzer/wiki/Server-setup#go-cryptotls-package

Reply:
perfect, thanks!